ir/type.def

```diff
@@ -493,16 +493,7 @@ class Type_SpecializedCanonical : Type {
         BUG_CHECK(substituted->is<IMayBeGenericType>(), "substituted %1% is not generic", substituted);
     }
     const Type* getP4Type() const override;
-    toString{
-        cstring result = baseType->toString() + "<";
-        cstring sep = "";
-        for (auto t : *arguments) {
-            result += sep + t->toString();
-            sep = ", ";
-        }
-        result += ">";
-        return result;
-    }
+    toString{ return baseType->toString() + "<...>"; }
```

Comment:
This change appears revert https://github.com/p4lang/p4c/commit/8eebedea818a2c788e0675d052a170be904a8e7f#diff-ea1da9796c6c2b602731a169985ebd06ed51a4b37dc65b5f5c8e67cba889c974R496

However, if we compare the output, the additional information about the arguments looks useful.

```diff
-issue3045-1.p4(16): Type of argument 'r' (register<...>) does not match type of parameter 'r' (register<...>)
+issue3045-1.p4(14): Type of argument 'r' (register<bool>) does not match type of parameter 'r' (register<T>)
```

Reply:
It looks useful, but it is sometimes wrong. When the compiler substitutes types this will print the substituted types instead of the original ones the programmer wrote. That's why I decided it's better this way. We had a similar issue with variable names, but that is easier to solve, since variables remember their original names, while other IR constructs don't.

Reply:
"wrong" meaning that it displays to the user types they never wrote in their program